packages/nimble-components/src/combobox/index.ts

```diff
@@ -159,11 +159,11 @@ export class Combobox
             const prevSelectedValue = this.options[this.selectedIndex]?.text;
             const nextSelectedValue = this.options[selectedIndex]?.text;
 
-            this.selectedIndex = prevSelectedValue !== nextSelectedValue
-                ? selectedIndex
-                : this.selectedIndex;
+            if (prevSelectedValue !== nextSelectedValue) {
+                this.selectedIndex = selectedIndex;
+            }
 
-            updatedValue = this.firstSelectedOption?.text || next;
+            updatedValue = this.firstSelectedOption?.text || updatedValue;
```

Comment:
This makes it clear that we're not actually changing the value if there's no `firstSelectedOption`.